File: 양승민/입문/src/lec12/java/lec12Main.java
package lec12.java;

public class lec12Main {
    public static void main(String[] args) {
        moveSomething(new Movable() {

            @Override
            public void move() {
                System.out.println("움직움직");
            }

            @Override
            public void fly() {
                System.out.println("날앙");
            }
        });
    }

    private static void moveSomething(Movable movable) {
        movable.fly();
        movable.move();
    }
}

File: 양승민/입문/src/lec12/kotlin/lec12Main.kt
package lec12.kotlin

import lec12.java.Movable

fun main() {
    moveSomething(object : Movable {
        override fun move() {
            println("움직움직")
        }

        override fun fly() {
            println("날앙")
        }
    })
}

private fun moveSomething(movable: Movable) {
    movable.move()
    movable.fly()
}